templates/zoo.cfg.j2

```diff
@@ -5,11 +5,7 @@ clientPort={{ client_port }}
 initLimit={{ init_limit }}
 syncLimit={{ sync_limit }}
 
-{#
-Also consider:
-server.{{server.id}}={{server.host}}:2888:3888
-#}
 
 {% for server in zookeeper_hosts %}
-server.{{loop.index}}={{server.host}}:2888:3888
+server.{{server.id}}={{server.host}}:2888:3888
```

Comment:
When discussing this change between Sumant and I, we could argue either way. Either use loop.index everywhere or use the server.id everywhere. But usage in zoo.cfg.j2
should match what is used in myid.j2 so that the identities are consistent.

The primary argument we could find for using server.id was that it gives an operator full control over the identity of the nodes, rather than imposing an identity bound to a list order which over the life of an ensemble might change for many reasons, none of which should impose a change to the identity of the members of the ensemble. _I'm not sure if this is a real risk or not_ but being conservative in our change suggested this course.

The risk with potentially passing groups['zookeepers'] is that this does not ensure consistent ordering from one run to the next (it's a dict, not a list), violating the principle of idempotency.  That issue is why things like this exist: https://github.com/openstack/openstack-ansible-plugins/blob/master/library/dist_sort FWIW.